pkg/snmp/snmp_test.go

```diff
@@ -160,6 +160,117 @@ snmp_listener:
 	assert.Equal(t, false, conf.Configs[2].CollectDeviceMetadata)
 }
 
+func TestNewNetworkDevicesListenerConfig(t *testing.T) {
```

Comment:
In general I don't think we should duplicate all tests to take into account both the new and the old config but instead move tests to be using the new config and add tests to check that the conversion is done accordingly.

If you feel like some tests do require testing both, we could try to refactor the test to loop through the two options (via table testing for instance, or defining sub-tests with an argument that you can call twice).

Reply:
I removed all the duplicated tests in [this commit](https://github.com/DataDog/datadog-agent/pull/24644/commits/111213efc988729971a4d6313d04df13943b0915)